x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/upgrade_prebuilt_rules_table/use_prebuilt_rules_upgrade_state.ts

```diff
@@ -50,6 +58,35 @@ export function usePrebuiltRulesUpgradeState(
     []
   );
 
+  // Implements concurrency control.
+  // Rule may be edited or a new prebuilt rules package version gets released.
+  // In any case current rule's `revision` or target rule's version
+  // will have higher values.
+  // Reset resolved conflicts in case of revision`s or version`s mismatch.
+  useEffect(() => {
+    for (const {
+      rule_id: ruleId,
+      current_rule: { revision: nextRevision },
+      target_rule: { version: nextVersion },
+    } of ruleUpgradeInfos) {
+      const cc = concurrencyControl.current[ruleId];
+      const hasInconsistency = cc && (cc.version !== nextVersion || cc.revision !== nextRevision);
+
+      // Reset rule's resolved conflicts
+      if (hasInconsistency) {
+        setRulesResolvedConflicts((prevRulesResolvedConflicts) => ({
+          ...prevRulesResolvedConflicts,
+          [ruleId]: {},
+        }));
+      }
```

Comment:
Can we add a toast message here to let the user know that the partial updates have been reset? Something like:
![image](https://github.com/user-attachments/assets/19d1a283-804b-48a1-9fc8-4224aa6d48d7)
I added an error toast as an example there, maybe a warning toast would be better.